fdbserver/RESTKmsConnector.actor.cpp

```diff
@@ -0,0 +1,1143 @@
+	}
+	if (refreshKmsUrls) {
+		validateKmsUrls(ctx);
+	}
+}
+
+void testParseKmsResponseFailure(Reference<RESTKmsConnectorCtx> ctx) {
+	Arena arena;
+	std::vector<EncryptCipherKeyDetails> cipherDetails;
+
+	// Case-I: Missing CipherDetails tag
+	{
+		rapidjson::Document doc;
+		doc.SetObject();
+
+		rapidjson::Value key(KMS_URLS_TAG, doc.GetAllocator());
+		rapidjson::Value refreshUrl;
+		refreshUrl.SetBool(true);
+		doc.AddMember(key, refreshUrl, doc.GetAllocator());
+
+		Reference<HTTP::Response> httpResp = makeReference<HTTP::Response>();
+		httpResp->code = HTTP::HTTP_STATUS_CODE_OK;
+		rapidjson::StringBuffer sb;
+		rapidjson::Writer<rapidjson::StringBuffer> writer(sb);
+		doc.Accept(writer);
+		httpResp->content.resize(sb.GetSize(), '\0');
+		memcpy(httpResp->content.data(), sb.GetString(), sb.GetSize());
+
+		try {
+			parseKmsResponse(ctx, httpResp, arena, cipherDetails);
+		} catch (Error& e) {
+			ASSERT_EQ(e.code(), error_code_operation_failed);
+		}
+	}
+	// Case-II: CipherDetails malformed (not an array)
+	{
+		rapidjson::Document doc;
+		doc.SetObject();
+
+		rapidjson::Value key(CIPHER_KEY_DETAILS_TAG, doc.GetAllocator());
+		rapidjson::Value details;
+		details.SetBool(true);
+		doc.AddMember(key, details, doc.GetAllocator());
+
+		Reference<HTTP::Response> httpResp = makeReference<HTTP::Response>();
+		httpResp->code = HTTP::HTTP_STATUS_CODE_OK;
+		rapidjson::StringBuffer sb;
+		rapidjson::Writer<rapidjson::StringBuffer> writer(sb);
+		doc.Accept(writer);
+		httpResp->content.resize(sb.GetSize(), '\0');
+		memcpy(httpResp->content.data(), sb.GetString(), sb.GetSize());
+
+		try {
+			parseKmsResponse(ctx, httpResp, arena, cipherDetails);
+		} catch (Error& e) {
+			ASSERT_EQ(e.code(), error_code_operation_failed);
+		}
+	}
+	// Case-III: Malformed CipherDetail object - missing encyrptDomainId/baseCipherId
+	{
+		rapidjson::Document doc;
+		doc.SetObject();
+
+		rapidjson::Value cDetails(rapidjson::kArrayType);
+		rapidjson::Value detail(rapidjson::kObjectType);
+		rapidjson::Value key(BASE_CIPHER_ID_TAG, doc.GetAllocator());
+		rapidjson::Value id;
+		id.SetUint(12345);
+		detail.AddMember(key, id, doc.GetAllocator());
+		cDetails.PushBack(detail, doc.GetAllocator());
+		key.SetString(CIPHER_KEY_DETAILS_TAG, doc.GetAllocator());
+		doc.AddMember(key, cDetails, doc.GetAllocator());
+
+		Reference<HTTP::Response> httpResp = makeReference<HTTP::Response>();
+		httpResp->code = HTTP::HTTP_STATUS_CODE_OK;
+		rapidjson::StringBuffer sb;
+		rapidjson::Writer<rapidjson::StringBuffer> writer(sb);
+		doc.Accept(writer);
+		httpResp->content.resize(sb.GetSize(), '\0');
+		memcpy(httpResp->content.data(), sb.GetString(), sb.GetSize());
+
+		try {
+			parseKmsResponse(ctx, httpResp, arena, cipherDetails);
+		} catch (Error& e) {
+			ASSERT_EQ(e.code(), error_code_operation_failed);
+		}
+	}
+}
+
+} // namespace
+
+TEST_CASE("fdbserver/RESTKmsConnector") {
```

Comment:
If each of these functions you're calling is testing separate functionality, which it appears to be, why not make them each their own test case?

Reply:
The test is intended to validate the RESTkmsConnector code that is "unit-testable", I could split it into multiple, but, then one needs to run multiple test which would be more work, but not much gain, imho.